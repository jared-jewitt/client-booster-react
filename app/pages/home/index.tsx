import React from "react";

import { AuthContext, IAuthContext } from "@/store";

import "./styles.scss";

export default (): JSX.Element => {
  const { clearAuthState, state } = React.useContext<IAuthContext>(AuthContext);

  return (
    <div className="home">
      <p>Welcome, {state.user?.name || "Unknown"}</p>
      <button onClick={() => clearAuthState()}>Simulate logout</button>
    </div>
  );
};
