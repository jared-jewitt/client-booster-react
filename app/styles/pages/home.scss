@use "../partials/placeholders/layout" as *;

.home {
  @extend %full-screen-center;
}
