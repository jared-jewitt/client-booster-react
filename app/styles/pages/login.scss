@use "../partials/placeholders/layout" as *;

.login {
  @extend %full-screen-center;

  &__logo {
    width: 200px;
    height: 200px;
    margin-bottom: 32px;
  }
}
