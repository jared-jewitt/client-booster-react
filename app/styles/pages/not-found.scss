@use "../partials/placeholders/layout" as *;

.not-found {
  @extend %full-screen-center;
}
