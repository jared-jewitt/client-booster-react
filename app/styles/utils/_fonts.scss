// Import fonts like such...
//@font-face {
//  font-family: 'OxygenRegular';
//  src: url('~@/assets/fonts/Oxygen-Regular.ttf') format('truetype');
//}
