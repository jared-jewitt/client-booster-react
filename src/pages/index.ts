export { default as Dashboard } from "@/pages/dashboard";
export { default as Login } from "@/pages/login";
