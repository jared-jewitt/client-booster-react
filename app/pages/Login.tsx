import React from "react";

import { AuthContext, IAuthContext } from "@/store";

import Logo from "@/assets/images/logo.svg";

import "@/styles/pages/login.scss";

export default (): JSX.Element => {
  const { updateAuthState } = React.useContext<IAuthContext>(AuthContext);

  return (
    <div className="login">
      <img className="login__logo" alt="Logo" src={Logo} />
      <button onClick={() => updateAuthState({ user: { name: "Foo Bar" } })}>Simulate login</button>
    </div>
  );
};
