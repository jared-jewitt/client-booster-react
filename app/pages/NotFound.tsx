import React from "react";

import "@/styles/pages/not-found.scss";

export default (): JSX.Element => (
  <div className="not-found">
    <p>404 - Not found</p>
  </div>
);
