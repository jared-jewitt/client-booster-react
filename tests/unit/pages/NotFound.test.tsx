import React from "react";

import { render } from "../../utils";

import { NotFoundPage } from "@/pages";

describe("NotFoundPage", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<NotFoundPage />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
