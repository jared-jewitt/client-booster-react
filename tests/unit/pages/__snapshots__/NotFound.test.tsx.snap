// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="not-found"
  >
    <p>
      404 - Not found
    </p>
  </div>
</DocumentFragment>
`;
